src/rust/ide/view/graph-editor/src/builtin/visualization/java_script.rs

```diff
@@ -20,14 +20,27 @@ pub fn table_view_visualization() -> visualization::java_script::FallibleDefinit
 
 /// Return a `JavaScript` Scatter plot visualization.
 pub fn scatter_plot_visualization() -> visualization::java_script::FallibleDefinition {
-    let source = include_str!("java_script/scatterPlot.js");
+    let loading_scripts = include_str!("java_script/loading.js");
+    let source          = include_str!("java_script/scatterPlot.js");
+    let source          = format!("{}{}",loading_scripts,source);
+
+    visualization::java_script::Definition::new(data::builtin_library(),source)
+}
+
+/// Return a `JavaScript` Histogram visualization.
+pub fn histogram_visualization() -> visualization::java_script::FallibleDefinition {
+    let loading_scripts = include_str!("java_script/loading.js");
+    let source          = include_str!("java_script/histogram.js");
+    let source          = format!("{}{}",loading_scripts,source);
 
     visualization::java_script::Definition::new(data::builtin_library(),source)
 }
 
 /// Return a `JavaScript` Map visualization.
 pub fn map_view_visualization() -> visualization::java_script::FallibleDefinition {
-    let source = include_str!("java_script/mapView.js");
+    let loading_scripts = include_str!("java_script/loading.js");
+    let source          = include_str!("java_script/mapView.js");
```

Comment:
I know its not part of this commit, but I dont like the name `mapView`. It should be `geoMap` instead. Every visualization is a "view", but we don't use the name "histogramView", nor other views.

Reply:
I'll do this as part of the map-initialisation PR.